bench/workgen/workgen.cxx

```diff
@@ -248,6 +281,231 @@ WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn)
+                    << _workload->options.dynamic_drop_trigger << " MB.");
+            }
+        }
+        if (!dropping) {
+            sleep(_workload->options.dynamic_drop_period);
+            continue;
+        }
+
+        for (int drops = 0; drops < _workload->options.dynamic_drop_count; ++drops) {
+            std::vector<std::string> drop_uris;
+            // The dynamic table data structures are protected by a mutex.
+            {
+                const std::lock_guard<std::mutex> lock(*icontext->_dyn_mutex);
+
+                // Walk the list and select the first that is not already in the delete list
+                for (auto const &it : icontext->_dyn_tint) {
+                    if (icontext->_dyn_table_runtime[it.second]._pending_delete) {
+                        continue;
+                    } else {
+                        VERBOSE(*_workload, "Marking pending removal for: " << it.first);
+                        icontext->_dyn_table_runtime[it.second]._pending_delete = true;
+                        tables_delete.push_back(it.first);
+                        break;
+                    }
+                }
+
+                /*
+                 * Process any pending deletes, the actual table drop will be done later without
+                 * holding the lock.
+                 */
+                for (size_t i = 0; i < tables_delete.size();) {
+                    // The table might still be in use.
+                    const std::string uri(tables_delete.at(i));
+                    tint_t tint = icontext->_dyn_tint.at(uri);
+                    if (icontext->_dyn_table_runtime[tint]._in_use != 0) {
+                        VERBOSE(*_workload, "Requested remove of table in use: " << uri);
+                        ++i;
+                        continue;
+                    }
+
+                    // Delete all local data related to the table.
+                    tables_delete.erase(tables_delete.begin() + i);
+                    icontext->_dyn_tint.erase(uri);
+                    icontext->_dyn_table_names.erase(tint);
+                    icontext->_dyn_table_runtime.erase(tint);
+                    drop_uris.push_back(uri);
+                }
+            }
+
+            /*
+             * We will drop the WiredTiger tables without holding the lock. These tables have been
+             * removed from the shared data structures, and we know no thread is operating on them.
+             */
+            for (auto uri : drop_uris) {
+                // Spin on EBUSY, we do not expect to get stuck
+                while ((ret = session->drop(session, uri.c_str(), "force,checkpoint_wait=false")) ==
+                  EBUSY) {
+                    VERBOSE(*_workload, "Drop returned EBUSY for table: " << uri);
+                }
+                if (ret != 0)
+                    THROW("Table drop failed in start_dynamic_table_mgmt.");
+
+                VERBOSE(*_workload, "Dropped table: " << uri);
```

Comment:
There's a lot of verbose output in this function. Is it all needed?

Reply:
As before, I have cleaned the verbose a bit, but this is very coarse compared to what already verbose does.